keras/layers/core.py

```diff
@@ -196,6 +198,11 @@ def get_output(self, train=False):
         elif self.mode == 'concat':
             inputs = [self.layers[i].get_output(train) for i in range(len(self.layers))]
             return T.concatenate(inputs, axis=self.concat_axis)
+        elif self.mode == 'join':
+            inputs = OrderedDict()
+            for i in range(len(self.layers)):
+                inputs[self.layers[i].get_output(train).name] = self.layers[i].get_output(train)
```

Comment:
What if the layer does not have a name, though? There should be an exception or a fallback (preferably an exception, for maximum explicitness).